src/LLVM_Output.cpp

```diff
@@ -87,9 +98,113 @@ void compile_llvm_module_to_llvm_assembly(llvm::Module &module, Internal::LLVMOS
     module.print(out, nullptr);
 }
 
-void create_static_library(const std::vector<std::string> &src_files, const Target &target,
-                    const std::string &dst_file, bool deterministic) {
-    internal_assert(!src_files.empty());
+// Note that the utilities for get/set working directory are deliberately *not* in Util.h;
+// generally speaking, you shouldn't ever need or want to do this, and doing so is asking for
+// trouble. This exists solely to work around an issue with LLVM, hence its restricted
+// location. If we ever legitimately need this elsewhere, consider moving it to Util.h.
+namespace {
+
+std::string get_current_directory() {
+#ifdef _WIN32
+    std::string dir;
+    char p[MAX_PATH];
+    DWORD ret = GetCurrentDirectoryA(MAX_PATH, p);
+    internal_assert(ret != 0) << "GetCurrentDirectoryA() failed";
+    dir = p;
+    return dir;
+#else
+    std::string dir;
+    char *p = getcwd(nullptr, 0);
```

Comment:
Might be worth a comment that we are intentionally deciding to use this not-100% standard behavior of getcwd. (Passing nullptr for the first argument is not guaranteed to work by POSIX, but it looks widely enough supported that it is the right thing to do.)